core-bundle/contao/dca/tl_undo.php

```diff
@@ -181,7 +182,7 @@ public function showDeletedRecords($data, $row)
 						}
 						else
 						{
-							$v = implode(', ', $array);
+							$v = ArrayUtil::implodeRecursive($array);
```

Comment:
How about using the Yaml dumper here?
The format should be very similar, but it supports all possible cases.

```suggestion
							$v = substr(Yaml::dump($array, 0), 1, -1);
```
and:
```
use Symfony\Component\Yaml\Yaml;
```

Reply:
Good idea, however, the output would be different:

<img width="877" alt="YAML" src="https://github.com/contao/contao/assets/1192057/f59503ca-82b7-40f5-bdb7-c6521a183eb2">

<img width="871" alt="ArrayUtil" src="https://github.com/contao/contao/assets/1192057/3a2544ef-0797-4412-a274-d37239a44960">

I don‘t know how important this is?

Reply:
Maybe this would fit better?

```suggestion
							$v = Yaml::dump($array, 1);
```

But that would also make simple arrays like `['foo', 'bar']` being multi-line then...

Reply:
Better: 

```suggestion
					        if (array_filter($v, static fn ($val) => \is_array($val))) {
					            $v = Yaml::dump($source, 1);
					        } else {
					            $v = substr(Yaml::dump($source, 0), 1, -1);
					        }
```

Reply:
This does make some views better, such as

<img width="424" alt="" src="https://github.com/contao/contao/assets/1192057/8e81d9a9-0e02-42fa-86fb-0ee197488409">

<img width="473" alt="" src="https://github.com/contao/contao/assets/1192057/1376d46e-f67c-45ac-8a31-65cd6252efe8">

---

But I‘m not sure if

<img width="874" alt="" src="https://github.com/contao/contao/assets/1192057/9f353922-b494-47fc-951a-b3f9693d9668">

is better (in terms of "more understandable") than

<img width="873" alt="" src="https://github.com/contao/contao/assets/1192057/e25cae2c-2801-4ed3-b749-57f3c7d8f480">

I‘ll have to think about it. 😄

Reply:
How about using `print_r()` and `white-space: pre-wrap`?

<img width="873" alt="" src="https://github.com/contao/contao/assets/1192057/cb8200fd-d8a0-4671-89bd-404446df69fd">

Reply:
Or `var_export()` with a monospace font?

<img width="867" alt="" src="https://github.com/contao/contao/assets/1192057/562e2ebd-dc63-40b3-a190-6e38643b8c8e">

Reply:
You could use the vardumper which is what we're doing for templates as well. With whatever styles you like but ut would give the ability to search, open and collapse variables and also takes care of formatting arbitrary values.

Reply:
> How about using `print_r()` and `white-space: pre-wrap`?

Then I would prefer `Yaml::dump($array, 99)` with `white-space: pre-wrap`